aea/helpers/dialogue/base.py

```diff
@@ -1076,46 +1086,15 @@ def _create(
         assert (
             dialogue_label not in self.dialogues
         ), "Dialogue label already present in dialogues."
-        if self._message_class is not None and self._dialogue_class is not None:
-            dialogue = self._dialogue_class(
-                dialogue_label=dialogue_label,
-                message_class=self._message_class,
-                agent_address=self.agent_address,
-                role=role,
-            )
-        else:
-            # TODO: remove this approach
-            dialogue = self.create_dialogue(
-                dialogue_label=dialogue_label, role=role,
-            )  # pragma: no cover
+        dialogue = self._dialogue_class(
+            dialogue_label=dialogue_label,
+            message_class=self._message_class,
+            agent_address=self.agent_address,
+            role=role,
+        )
         self.dialogues.update({dialogue_label: dialogue})
         return dialogue
 
-    @abstractmethod
-    def create_dialogue(
-        self, dialogue_label: DialogueLabel, role: Dialogue.Role,
-    ) -> Dialogue:
-        """
-        THIS METHOD IS DEPRECATED AND WILL BE REMOVED IN THE NEXT VERSION. USE THE NEW CONSTRUCTOR ARGUMENTS INSTEAD.
-
-        Create a dialogue instance.
-
-        :param dialogue_label: the identifier of the dialogue
-        :param role: the role of the agent this dialogue is maintained for
-
-        :return: the created dialogue
-        """
-
-    @staticmethod
-    def role_from_first_message(message: Message) -> Dialogue.Role:
-        """
-        Infer the role of the agent from an incoming or outgoing first message.
-
-        :param message: an incoming/outgoing first message
-        :return: the agent's role
-        """
-        pass  # pragma: no cover
```

Comment:
ensure dynamically provided method has this signature

Reply:
See above.